es6/docxtemplater.js

```diff
@@ -28,15 +28,22 @@ const {
 const currentModuleApiVersion = [3, 21, 0];
 
 const Docxtemplater = class Docxtemplater {
-	constructor() {
-		if (arguments.length > 0) {
+	constructor(zip, { modules = [] } = {}) {
+		if (!Array.isArray(modules)) {
 			throw new Error(
-				"The constructor with parameters has been removed in docxtemplater 3, please check the upgrade guide."
+				"Please pass modules as an array"
 			);
 		}
```

Comment:
Can you add a condition to check that if the first arg is set, it is a valid jszip.

You can do that with following condition : 

`if (zip.files && typeof zip.file === "function")`

If it is not a zip file, can you throw the following error : 

```
throw new Error("The first argument of docxtemplater's constructor is not a valid zip file")
```

Reply:
I thought about checking for the zip but decided to go against it because it was already being checked in loadZip() method. Anyways, Its never a bad idea to be explicit about an error. I will do it.